frontend/src/entry/overview/FormEditPage.tsx

```diff
@@ -142,6 +143,7 @@ export default function FormEditPage({ entry }: FormEditPageProps) {
                 onClick={() => setIsEdit(!isEdit)}
                 sx={{ mb: { xs: 2 } }}
                 data-test='editEntryCardButton'
+                hidden={readOnly}
```

Comment:
Again, we should be conditionally rendering this button.

However, due to your conditional logic in `frontend/src/entry/overview/Overview.tsx` I don't think we need to do hide anything in this component.